Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
+				{
+					uint ids_name;
+					wstring nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
+						{
+							ids_name = ini.get_value_int(0);
+							ShipnamesWhichAlreadyUsedResetRep[nickname] = ids_name;
+						}
+					}
+					
+				}
+			}
+			ini.close();
+			ConPrint(L"Rep: Loaded %u factions\n", factions.size());
+		}
+	}
+
+	void MiscCmds::Resetrep_Time_limits_save()
+	{
+		string factionpropfile = resetrepfileadress;
+
+		FILE *file = fopen(factionpropfile.c_str(), "w");
+		if (file)
+		{
+			fprintf_s(file, "[Playerships]\n");
+			for (map<wstring, uint>::iterator i = ShipnamesWhichAlreadyUsedResetRep.begin();
+				i != ShipnamesWhichAlreadyUsedResetRep.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
+				{
+					wstring temp = i->first;
+					ini_write_wstring(file, "shipname", temp);
+					fprintf_s(file, "timelimit = %d\n", i->second);
+				}
+			}
+			fclose(file);
+		}
+	}
+
+	/** A command to reset all reputations to zero that you might have */
+	bool MiscCmds::UserCmd_ResetRep(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+	{
+		HK_ERROR err;
+
+		wstring wscCharname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+
+		uint iShip;
+		pub::Player::GetShip(iClientID, iShip);
+		if (iShip)
+		{
+			PrintUserCmdText(iClientID, L"ERR Not in base");
+			return true;
+		}
+
+		// Read the current number of credits for the player
+		// and check that the character has enough cash.
+		int iCash = 0;
+		if ((err = HkGetCash(wscCharname, iCash)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR %s", HkErrGetText(err).c_str());
+			return true;
+		}
+
+		if (set_iResetdropCost > 0 && iCash < set_iResetdropCost)
+		{
+			PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+			return true;
+		}
+
+
+		wstring wscCharFileName;
+		if ((err = HkGetCharFileName(wscCharname, wscCharFileName)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR " + HkErrGetText(err));
+			return true;
+		}
+
+		if (ShipnamesWhichAlreadyUsedResetRep.find(wscCharFileName) != ShipnamesWhichAlreadyUsedResetRep.end())
+		{
+			uint LastTime = ShipnamesWhichAlreadyUsedResetRep[wscCharFileName];
+
+			if (((int)time(0) - LastTime) < set_iResetrepTimeLimit)
+			{
+				int SecondsLeft = (set_iResetrepTimeLimit - ((int)time(0) - int(LastTime)));
+				int DaysLeft = SecondsLeft / (3600 * 24);
+			PrintUserCmdText(iClientID, L"ERR Resetrep time limit. %d days (%d in seconds) time remains to wait.", DaysLeft, SecondsLeft);
+			return true;
+			}
+		}
+
+		int count = 0;
+		for (map<string, uint>::iterator iter = factions.begin(); iter != factions.end(); iter++)
+		{
+			float fRep = 0.0f;
+			string factionName = iter->first;
+			HK_ERROR error;
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
```

Comment:
what purpose does all this serve exactly?
do we want to check fRep != 0.0f before count++?

Reply:
Just a check for possible errors. To verify that every used factionName belongs to some registered faction.
If it would not pass, usage of HkSetRep can be not recommended... for more stable command work probably will be better having "continue;"  instead of "return true" in error catching if construction. Changed to "continue" to skip current for loop step.

Reply:
Given that this is data from initialworld isn't this effectively dead code that can never be hit?

Reply:
Yes, but.... moderators can mess up this data with being still working
This check makes sure that "SetRep" is not applied to entity which can't be changed
saving the game from possible crash to just skipped loop cycle.

This check is used very rarely anyway, only when player uses the command
So it should be as crash-proof as possible in my opinion.